docs/source/community/design.rst

```diff
@@ -0,0 +1,158 @@
+often used interchangeably in everyday English. Consider how one may
+model `devices <https://pytorch.org/docs/master/tensor_attributes.html#torch.device>`__
+in PyTorch:
+
+-  **Simple / Explicit (to understand, debug):** every tensor is associated
+   with a device. The user explicitly specifies tensor device movement.
+   Operations that require cross-device movement result in an error.
+-  **Easy / Implicit (to use):** the user does not have to worry about
+   devices; the system figures out the globally optimal device
+   placement.
+
+In this specific case, and as a general design philosophy, PyTorch
+favors exposing simple and explicit building blocks rather than APIs
+that are easy-to-use by practitioners. The simple version is immediately
+understandable and debuggable by a new PyTorch user: you get a clear
+error if you call an operator requiring cross-device movement at the
+point in the program where the operator is actually invoked. The easy
+solution may let a new user move faster initially, but debugging such a
+system can be complex: How did the system make its determination? What
+is the API for plugging into such a system and how are objects
+represented in its IR?
+
+Some classic arguments in favor of this sort of design come from `A
+Note on Distributed
+Computation <https://dl.acm.org/doi/book/10.5555/974938>`__ (TLDR: Do not
+model resources with very different performance characteristics
+uniformly, the details will leak) and the `End-to-End
+Principle <http://web.mit.edu/Saltzer/www/publications/endtoend/endtoend.pdf>`__
+(TLDR: building smarts into the lower-layers of the stack can prevent
+building performant features at higher layers in the stack, and often
+doesn’t work anyway). For example, we could build operator-level or
+global device movement rules, but the precise choices aren’t obvious and
+building an extensible mechanism has unavoidable complexity and latency
+costs.
+
+A caveat here is that this does not mean that higher-level “easy” APIs
+are not valuable; certainly there is a value in, for example,
+higher-levels in the stack to support efficient tensor computations
+across heterogeneous compute in a large cluster. Instead, what we mean
+is that focusing on simple lower-level building blocks helps inform the
+easy API while still maintaining a good experience when users need to
+leave the beaten path. It also allows space for innovation and the
+growth of more opinionated tools at a rate we cannot support in the
+PyTorch core library, but ultimately benefit from, as evidenced by
+our `rich ecosystem <https://pytorch.org/ecosystem/>`__. In other
+words, not automating at the start allows us to potentially reach levels
+of good automation faster.
+
+Principle 3: Python First with Best In Class Language Interoperability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This principle began as `Python
+First <http://This%20link%20seems%20broken>`__: 
```

Comment:
This link was broken in the Google doc. Can we remove it and just keep "Python First" w/o link?

Reply:
Agreed - let's do that. Thanks!